pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -639,4 +642,36 @@ func init() {
 		panic(err)
 	}
 
+	err = InfraSAs.addServiceAccount(
+		InfraEndpointControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: EndpointControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// Watching services and pods
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("get", "list", "watch"),
+					Resources: sets.NewString("services", "pods"),
+				},
+				// Managing endpoints
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("get", "list", "create", "update", "delete"),
+					Resources: sets.NewString("endpoints"),
+				},
+				// Permission for RestrictedEndpointsAdmission
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("create"),
+					Resources: sets.NewString("endpoints/restricted"),
+				},
+			},
```

Comment:
I don't think so? There's nothing relevant about "events" in endpoints_controller.go, and it doesn't log any errors with the current set of permissions.